### PR TITLE
Added PercentPoints to Dimension

### PR DESCRIPTION
**Pull request by not-aidan**
# Objective

Why did you make this PR?
See #229

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Very nicely done, especially for a first contribution! Can you add a note to RELEASES.md?

File: src/style.rs [hunk 1]
```diff
@@ -266,6 +266,8 @@ pub enum Dimension {
     Points(f32),
     /// The dimension is stored in percentage relative to the parent item.
     Percent(f32),
+    /// The dimension is stored as the sum of percentage relative to the parent item and point.
```
Comment (alice-i-cecile) [inline src/style.rs:269]: I think we should add a little bit more color to the docs here, explaining why this is useful (and when you might want to use it).

File: src/style.rs [hunk 2]
```diff
@@ -266,6 +266,8 @@ pub enum Dimension {
...
+    PercentPoints(f32, f32),
```
Comment (alice-i-cecile) [inline src/style.rs:270]: I would be more comfortable if this was a named; tuple types that contain the same internal value are a recipe for hard-to-spot bugs.

**Comment by alice-i-cecile** [comment]
Can you please rebase? This PR is quite hard to review in its current form.

**Comment by nicoburns** [comment]
@alice-i-cecile All open PRs that were opened before the recent force-push to main will need to be rebased, and it's not an entirely simple process, so I'm posting instructions:

First ensure you have have the latest main (`git pull` will *not* work)
```bash
git fetch
git checkout main
git reset --hard origin/main
```
If you want you can also save the old main to a branch `old-main`:
```bash
git checkout -b old-main
git reset --hard d295d9b5a1de28fa6644a7a4d5fefcd3b6957773
```
(`d295d9b5a1de28fa6644a7a4d5fefcd3b6957773` happens to be the commit ID of the latest commit to main pre-rebase)

### Simple case: branch rebasing cleanly on latest main:

1. Rebase directly onto new main

```
git checkout pr-branch
git rebase --onto main d295d9b5a1de28fa6644a7a4d5fefcd3b6957773
```

### Complex case: branch already had conflicts prior to force-push

1. Find the branch point of the PR branch with the old main branch
```bash
git merge-base pr-branch-name d295d9b5a1de28fa6644a7a4d5fefcd3b6957773
```
Save this "old main branch point" commit id for later.

2. Find the equivalent commit in the new main branch
```bash
git log --grep="$(git log --pretty=format:"%s" old-main-branch-point | head -n 1)" main
```
where old-main-branch-point is commit id from step 3.
Save this "new main branch point" commit id for later

3. Rebase the branch

```
git checkout pr-branch
git rebase --onto new-branch-point old-branch-point
```

**Comment by alice-i-cecile** [comment]
Thank you very much @nicoburns, those instructions will be very helpful.

**Comment by alice-i-cecile** [comment]
Closing this out due to merge conflicts. This can be remade and resubmitted if you would still like this feature.

**Pull request closed without merging**